source/concepts.tex

```diff
@@ -0,0 +1,1249 @@
+\end{note}
+
+\end{itemdescr}
+
+\rSec2[concepts.lib.corelang.integral]{Concept \tcode{Integral}}
+
+\indexlibrary{\idxcode{Integral}}%
+\begin{itemdecl}
+template <class T>
+concept Integral = is_integral_v<T>;
+\end{itemdecl}
+
+\begin{itemdescr}
+\pnum
+There need be no subsumption relationship between \tcode{Integral<T>} and
+\tcode{is_integral_v<T>}.
+\end{itemdescr}
+
+\rSec2[concepts.lib.corelang.signedintegral]{Concept \tcode{SignedIntegral}}
+
+\indexlibrary{\idxcode{SignedIntegral}}%
+\begin{itemdecl}
+template <class T>
+concept SignedIntegral = Integral<T> && is_signed_v<T>;
+\end{itemdecl}
+
+\begin{itemdescr}
+\pnum
+There need be no subsumption relationship between \tcode{SignedIntegral<T>} and
+\tcode{is_signed_v<T>}.
+
+\pnum
+\begin{note}
+\tcode{SignedIntegral<T>} can be satisfied even by types that are not signed
+integral types\iref{basic.fundamental}; for example, \tcode{char}.
+\end{note}
+\end{itemdescr}
+
+\rSec2[concepts.lib.corelang.unsignedintegral]{Concept \tcode{UnsignedIntegral}}
+
+\indexlibrary{\idxcode{UnsignedIntegral}}%
+\begin{itemdecl}
+template <class T>
+concept UnsignedIntegral = Integral<T> && !SignedIntegral<T>;
+\end{itemdecl}
+
+\begin{itemdescr}
+\pnum
+\begin{note}
+\tcode{UnsignedIntegral<T>} can be satisfied even by types that are not unsigned
+integral types\iref{basic.fundamental}; for example, \tcode{char}.
+\end{note}
+\end{itemdescr}
+
+\rSec2[concepts.lib.corelang.assignable]{Concept \tcode{Assignable}}
+
+\indexlibrary{\idxcode{Assignable}}%
+\begin{itemdecl}
+template <class LHS, class RHS>
+concept Assignable =
+  is_lvalue_reference_v<LHS> &&
+  CommonReference<const remove_reference_t<LHS>&, const remove_reference_t<RHS>&> &&
+  requires(LHS lhs, RHS&& rhs) {
+    lhs = std::forward<RHS>(rhs);
+    requires Same<decltype(lhs = std::forward<RHS>(rhs)), LHS>;
+  };
+\end{itemdecl}
+
+\begin{itemdescr}
+\pnum
+Let:
+\begin{itemize}
+\item \tcode{lhs} be an lvalue that refers to an object \tcode{lcopy} such that
+  \tcode{decltype((lhs))} is \tcode{LHS},
+\item \tcode{rhs} be an expression such that \tcode{decltype((rhs))} is \tcode{RHS}, and
+\item \tcode{rcopy} be a distinct object that is equal to \tcode{rhs}.
+\end{itemize}
+\tcode{Assignable<LHS, RHS>} is satisfied only if
+
+\begin{itemize}
+\item \tcode{addressof(lhs = rhs) == addressof(lcopy)}.
+
+\item After evaluating \tcode{lhs = rhs}:
+
+\begin{itemize}
+\item \tcode{lhs} is equal to \tcode{rcopy}, unless \tcode{rhs} is a non-const
+xvalue that refers to \tcode{lcopy}.
+
+\item If \tcode{rhs} is a non-\tcode{const} xvalue, the resulting state of the
+object to which it refers is valid but unspecified\iref{lib.types.movedfrom}.
+
+\item Otherwise, if \tcode{rhs} is a glvalue, the object to which it refers is not
+modified.
+\end{itemize}
+\end{itemize}
+
+\pnum
+There need be no subsumption relationship between \tcode{Assignable<LHS, RHS>}
+and \tcode{is_\-lval\-ue_\-ref\-er\-ence_v<LHS>}.
```

Comment:
Could you please limit the hyphenation hint to exactly the one you need to avoid the overfull \hbox, if any?